Handle proxy related env vars in the Operator

### Description

The Operator should correctly handle proxy related env vars: `HTTP_PROXY`, `HTTPS_PROXY` and `NO_PROXY`. Those env vars are de facto a [standard](https://sdk.operatorframework.io/docs/building-operators/golang/references/proxy-vars/).

The Operator should take one of the following approaches:
* Ignore the env vars as it might not make much sense for the Operator to use proxy. Only outgoing requests are toward the Kube API.
* Properly support the env vars and use the specified proxy setting for all potential outgoing requests (incl. Kube API).
  * A best practice is also considered for the Operator to pass down the env vars to the Operand pods.

### Discussion

_No response_

### Motivation

Currently, it looks like the env vars are recognized by Fabric8, however the implementation might suffer from some issues. As reported for RHBK, an exception is thrown when the proxy env vars do not include a port:
```
ERROR [io.qua.run.Application] (main) Failed to start application (with profile [prod]): java.lang.RuntimeException: Failed to start quarkus
    at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
    at io.quarkus.runtime.Application.start(Application.java:101)
    at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
    at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
    at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
    at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
    at io.quarkus.runner.GeneratedMain.main(Unknown Source)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.base/java.lang.reflect.Method.invoke(Method.java:568)
    at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:62)
    at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:33)
Caused by: java.lang.IllegalArgumentException: Failure in creating proxy URL. Proxy port is required!
    at io.fabric8.kubernetes.client.utils.HttpClientUtils.getProxyUri(HttpClientUtils.java:91)
    at io.fabric8.kubernetes.client.utils.HttpClientUtils.configureProxy(HttpClientUtils.java:210)
```
This is essentially a bug but since the Operator doesn't claim support for the proxy env vars, tracking this as an enhancement.

This behaviour is particularly problematic when a cluster-wide proxy is configured and the env vars implicitly set on the pods. In this case it might be problematic for the user to overwrite the env vars.

### Details

Seems to be related to: https://github.com/fabric8io/kubernetes-client/issues/3671

--- a/operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentDependentResource.java
+++ b/operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentDependentResource.java
@@ -427,7 +427,7 @@ public class KeycloakDeploymentDependentResource extends CRUDKubernetesDependent
         for (String env : COPY_ENV) {
             String value = System.getenv(env);
             if (value != null) {
-                envVars.add(new EnvVar(env, value, null));
+                envVars.add(new EnvVarBuilder().withName(env).withValue(value).build());
             }
         }
 

--- a/operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentDependentResource.java
+++ b/operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentDependentResource.java
@@ -53,6 +53,7 @@ import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Base64;
 import java.util.LinkedHashMap;
 import java.util.List;
@@ -73,6 +74,8 @@ import static org.keycloak.operator.crds.v2alpha1.CRDUtils.isTlsConfigured;
 
 @KubernetesDependent(labelSelector = Constants.DEFAULT_LABELS_AS_STRING)
 public class KeycloakDeploymentDependentResource extends CRUDKubernetesDependentResource<StatefulSet, Keycloak> {
+
+    private static final List<String> COPY_ENV = Arrays.asList("HTTP_PROXY", "HTTPS_PROXY", "NO_PROXY");
 
     private static final String SERVICE_ACCOUNT_DIR = "/var/run/secrets/kubernetes.io/serviceaccount/";
     private static final String SERVICE_CA_CRT = SERVICE_ACCOUNT_DIR + "service-ca.crt";
@@ -419,7 +422,14 @@ public class KeycloakDeploymentDependentResource extends CRUDKubernetesDependent
                     }
                     return envBuilder.build();
                 })
-                .collect(Collectors.toList());
+                .collect(Collectors.toCollection(ArrayList::new));
+
+        for (String env : COPY_ENV) {
+            String value = System.getenv(env);
+            if (value != null) {
+                envVars.add(new EnvVar(env, value, null));
+            }
+        }
 
         envVars.add(
                 new EnvVarBuilder()

